### PR TITLE
Add an API isinstance(inst, cls) that mirrors the Python isinstance.

### PR DESCRIPTION
**Pull request by hawkinsp**
The current `isinstance<T>(inst)` tests if a Python object is an instance of a bound C++ class. This new overload tests if a Python object is an instance of a Python class. I have needed this helper in several nanobind-using codebases, and the same API exists in pybind11.

### CODE REVIEW

File: include/nanobind/nb_misc.h [hunk 1]
```diff
@@ -111,6 +111,13 @@ inline Py_hash_t hash(handle h) {
     return rv;
 }
 
+inline bool isinstance(handle inst, handle cls) {
+  int ret = PyObject_IsInstance(inst.ptr(), cls.ptr());
```
Comment (wjakob) [inline include/nanobind/nb_misc.h:115]: indentation is different

**Comment by wjakob** [comment]
This looks great. Could I ask you to add a changelog entry?

**Comment by hawkinsp** [comment]
> This looks great. Could I ask you to add a changelog entry?

Done!

**Comment by wjakob** [comment]
Thanks!

**Pull request merged**